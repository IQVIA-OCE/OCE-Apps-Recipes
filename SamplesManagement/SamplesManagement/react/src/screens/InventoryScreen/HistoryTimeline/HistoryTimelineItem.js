import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Text, useTheme } from '@oce-apps/apollo-react-native';
import recordTypeIcons from './RecordTypeIcons';

import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import color from 'color';

const defaultTextColor = '#080707';

const HistoryTimelineItem = ({ item }) => {
  const icon = recordTypeIcons[item.recordType.DeveloperName];
  const theme = useTheme();
  const textColor = theme.dark ? color(defaultTextColor).negate().hex() : defaultTextColor;

  return (
    <View style={styles.container}>
      <View style={styles.leftSide}>
        <View style={styles.iconContainer}>
          <View style={{ ...styles.iconBg, backgroundColor: icon.color }}>
            <Icon name={icon.name} size={17} color="white" />
          </View>
          <View style={{ ...styles.line, backgroundColor: icon.color }} />
        </View>
        <View style={styles.detailsContainer}>
          <Text numberOfLines={1} style={[styles.recordNameText, { color: textColor }]}>
            {item.recordType.Name}
          </Text>
          <Text numberOfLines={2} style={[styles.productName, { color: textColor }]}>
            {item.productName}
          </Text>
          <Text style={styles.createdDate}>{item.transactionDateTime}</Text>
        </View>
      </View>
      <View style={styles.rightSide}>
        <Text style={styles.lotId}>{item.lotNumber}</Text>
        <View
          style={{ borderRadius: 50, backgroundColor: '#f88962', padding: 2 }}
        >
          <Text style={{ color: 'white', fontWeight: 'bold' }}>
            {item.quantity}
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  leftSide: {
    flexDirection: 'row',
    flexGrow: 1,
    flexBasis: 0,
  },
  iconContainer: {
    width: 24,
    marginRight: 12,
    alignItems: 'center',
  },
  iconBg: {
    padding: 3,
    borderRadius: 4,
    width: 24,
    height: 24,
    alignItems: 'center',
    alignContent: 'center',
    marginBottom: 2,
  },
  line: {
    flex: 1,
    width: 2,
    backgroundColor: 'grey',
  },
  recordNameText: {
    fontSize: 13,
    marginBottom: 5,
    fontWeight: '400',
  },
  productName: {
    fontSize: 13,
    marginBottom: 5,
    fontWeight: '400',
  },
  createdDate: {
    fontSize: 11,
    marginBottom: 24,
    fontWeight: '400',
    color: '#3e3e3c',
  },
  rightSide: {
    alignItems: 'flex-end',
    flexGrow: 1,
    flexBasis: 0,
  },
  lotId: {
    fontSize: 12,
    color: '#3e3e3c',
  },
});

export default HistoryTimelineItem;
