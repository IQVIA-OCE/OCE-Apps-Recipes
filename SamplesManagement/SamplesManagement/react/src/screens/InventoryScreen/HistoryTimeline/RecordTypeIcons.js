export default {
  AcknowledgementOfShipment: {
    name: 'format-list-checks',
    color: '#ef6e64',
  },
  Adjustment: {
    name: 'tune-vertical',
    color: '#47cfd2',
  },
  Disbursement: {
    name: 'file-document-box-outline',
    color: '#769ed9',
  },
  TransferIn: {
    name: 'format-indent-increase',
    color: '#f88962',
  },
  TransferOut: {
    name: 'format-indent-decrease',
    color: '#89c650',
  },
  Order: {
    name: 'clipboard-check',
    color: '#69cd37',
  },
  Return: {
    name: 'package-variant-closed',
    color: '#b781d3',
  },
};
