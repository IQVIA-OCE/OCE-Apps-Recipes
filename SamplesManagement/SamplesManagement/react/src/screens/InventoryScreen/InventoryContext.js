import React, { useContext, useEffect, useState } from 'react';
import {
  fetchActiveLotsProducts,
  fetchConfigs,
  fetchInventoryById,
  fetchInventoryDetail,
  fetchInventoryTypes,
  fetchLastSubmittedInventory,
  fetchTransactionDetails,
} from '../../api/Inventories';
import { normalizer } from '../../utils/utils';
import { AppContext } from '../../../AppContext';
import {
  getConfig,
  normalizeProductsList,
  normalizeProductsHistoryList,
} from './utils';
import { withNavigation } from 'react-navigation';
import { INVENTORY_STATUS } from './constants';
import { normalizeRecordTypes } from '../DashboardScreen/InventoriesWidget/utils';
import moment from 'moment';
import { INVENTORY_FORM_TYPE } from '../../constants/Inventories';

const initialValue = {
  editingType: INVENTORY_FORM_TYPE.preview,
  isLoading: true,
  preselectedProducts: [],
  products: [],
  productsHistory: [],
  error: null,
  config: {},
  lastInventoryCreatedDate: null,
  transactionDetails: [],
  recordType: {},
  formInitialValues: {},
};
export const InventoryContext = React.createContext(initialValue);

const inventoryKeys = {
  CreatedDate: 'createdDate',
  Id: 'id',
  OCE__Status__c: 'status',
  RecordTypeId: 'recordTypeId',
  OCE__Comments__c: 'comments',
  Name: 'name',
  OCE__Reason__c: 'reason',
  OCE__Auditor__c: 'auditor',
};

const configKeys = {
  OCE__SIShowSystemCalculatedFields__c: 'showCalculatedFields',
  OCE__SIShowSystemCount__c: 'showSystemCount',
  SetupOwnerId: 'ownerId',
  OCE__SIShowLoggedInUserRecords__c: 'record',
  OCE__STFinalStatus__c: 'status',
  OCE__SIHistoryListHidden__c: 'historyHidden',
};

const ContextProvider = ({ children, navigation }) => {
  const [state, setState] = useState(initialValue);
  const { userId, profileId, orgId } = useContext(AppContext);

  const handleFetchConfig = async () => {
    const [configsData] = await fetchConfigs(userId, profileId, orgId);

    const configs = normalizer(configKeys)(configsData);
    return getConfig(configs, [userId, profileId, orgId]);
  };

  const handleEditingType = editingType =>
    setState(prevState => ({ ...prevState, editingType }));

  useEffect(() => {
    handlePrepareData();
  }, []);

  const handlePrepareData = async () => {
    try {
      let inventory,
        transactionDetails,
        lastSubmittedInventoriesDetails,
        inventoryDetails,
        products,
        preselectedProducts,
        lastInventoryCreatedDate;
      const config = await handleFetchConfig();

      const [recordTypesData] = await fetchInventoryTypes();
      const recordTypes = normalizeRecordTypes(recordTypesData);

      const [lastInventories] = await fetchLastSubmittedInventory(
        userId,
        config.record
      );

      if (lastInventories.length) {
        inventory = [lastInventories[lastInventories.length - 1]];

        lastInventoryCreatedDate = inventory[0].CreatedDate;
        [lastSubmittedInventoriesDetails] = await fetchInventoryDetail(
          inventory[0].Id,
          true
        );
      } else {
        inventory = [
          {
            status: INVENTORY_STATUS.inProgress,
          },
        ];
      }

      if (navigation.state.params.id) {
        // preview or edit existing inventory
        [inventory] = await fetchInventoryById(navigation.state.params.id);
        [inventoryDetails] = await fetchInventoryDetail(
          navigation.state.params.id,
          state.editingType === INVENTORY_FORM_TYPE.preview
            ? false
            : inventory[0].OCE__Status__c !== INVENTORY_STATUS.inProgress
        );
      }

      [inventory] = normalizer(inventoryKeys)(inventory);

      if (inventory.status === INVENTORY_STATUS.inProgress) {
        [transactionDetails] = await fetchTransactionDetails(
          null,
          userId,
          config.status || 'Submitted'
        );
      } else {
        [transactionDetails] = await fetchTransactionDetails(
          navigation.state.params.id
        );
      }
      let productsHistory = normalizeProductsHistoryList(transactionDetails);

      const [productsList] = await fetchActiveLotsProducts();

      [products, preselectedProducts] = normalizeProductsList(
        productsList,
        lastSubmittedInventoriesDetails,
        inventoryDetails,
        transactionDetails,
        navigation.state.params.id ? inventory.status : null,
        Boolean(navigation.state.params.id)
      );

      setState(prevState => {
        let state = {
          ...prevState,
          isLoading: false,
          products,
          productsHistory,
          config,
          transactionDetails,
          lastInventoryCreatedDate,
        };
        if (navigation.state.params.id) {
          const recordType = recordTypes.all.find(
            el => el.Id === inventory.recordTypeId
          );

          state.editingType =
            navigation.state.params.type || INVENTORY_FORM_TYPE.preview;
          state.recordType = recordType;
          state.formInitialValues = {
            ...inventory,
            datetime: inventory.createdDate,
            products: preselectedProducts,
            buttonPressed: '',
          };
        } else {
          state.editingType = INVENTORY_FORM_TYPE.edit;
          state.recordType = navigation.state.params.recordType;
          state.formInitialValues = {
            id: '',
            datetime: moment(),
            products: preselectedProducts,
            status: INVENTORY_STATUS.inProgress,
            buttonPressed: '',
          };
        }

        return state;
      });
    } catch (error) {
      console.warn({ error });
      setState(prevState => ({
        ...prevState,
        isLoading: false,
        error,
      }));
    }
  };

  return (
    <InventoryContext.Provider
      value={{
        ...state,
        actions: { handleEditingType },
      }}
    >
      {children}
    </InventoryContext.Provider>
  );
};

export default withNavigation(ContextProvider);
