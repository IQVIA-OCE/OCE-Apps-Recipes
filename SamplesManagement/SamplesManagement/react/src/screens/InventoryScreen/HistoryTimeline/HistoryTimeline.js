import React, { useContext } from 'react';
import { StyleSheet, View, Text, FlatList } from 'react-native';
import { InventoryContext } from '../InventoryContext';
import HistoryTimelineItem from './HistoryTimelineItem';
import {
  secondaryBlue,
  black,
  IconButton,
  themeGrey,
} from 'apollo-react-native';

const HistoryTimeline = ({ product, clearSelectedProduct }) => {
  const { productsHistory } = useContext(InventoryContext);
  const filteredHistory = product
    ? productsHistory.filter(
        productHistory =>
          productHistory.productId === product.sampleProductId &&
          productHistory.lotId === product.lotNumberId
      )
    : productsHistory;

  return (
    <View style={styles.container}>
      {product ? (
        <View style={styles.selectedProduct}>
          <View>
            <Text style={{ color: black[300] }}>
              {product.label} ( {product.detailLabel} )
            </Text>
          </View>
          <View>
            <IconButton
              icon="close"
              color={black[500]}
              size={20}
              onPress={clearSelectedProduct}
            />
          </View>
        </View>
      ) : null}

      <FlatList
        data={filteredHistory}
        style={styles.list}
        keyExtractor={(item, i) => `${i}`}
        renderItem={({ item }) => <HistoryTimelineItem item={item} />}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    paddingHorizontal: 5,
    borderRightColor: themeGrey[200],
    borderRightWidth: 1,
  },
  list: {
    flexGrow: 1,
    flexBasis: 0,
    padding: 10,
  },
  titleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 5,
  },
  title: {
    fontSize: 17,
  },
  icon: {
    borderRadius: 4,
    width: 22,
    height: 22,
    alignItems: 'center',
    justifyContent: 'center',
    paddingTop: 1,
    marginRight: 10,
    backgroundColor: secondaryBlue[200],
  },
  selectedProduct: {
    marginHorizontal: 15,
    paddingBottom: 10,
    borderBottomColor: '#d8dde6',
    borderBottomWidth: 1,
    flexDirection: 'row',
    alignContent: 'center',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});

export default HistoryTimeline;
