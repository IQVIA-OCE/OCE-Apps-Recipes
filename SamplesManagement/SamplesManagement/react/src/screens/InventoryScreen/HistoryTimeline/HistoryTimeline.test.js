import React from 'react';
import { InventoryContext } from '../InventoryContext';
import HistoryTimeline from './HistoryTimeline';
import { render } from '@testing-library/react-native';

const product = {
  sampleProductId: '1',
  lotNumberId: '1',
  label: 'label',
  detailLabel: 'detailLabel',
};

const item = {
  productId: '1',
  lotId: '1',
  recordType: {
    DeveloperName: 'AcknowledgementOfShipment',
    Name: '',
  },
  productName: '',
  transactionDateTime: '',
  lotNumber: '',
  quantity: '',
};

describe('HistoryTimeline', () => {
  it('should render properly', () => {
    const { queryByText, rerender } = render(
      <InventoryContext.Provider value={{ productsHistory: [item] }}>
        <HistoryTimeline product={product} />
      </InventoryContext.Provider>
    );

    expect(queryByText(/detailLabel/)).toBeTruthy();

    rerender(
      <InventoryContext.Provider value={{ productsHistory: [item] }}>
        <HistoryTimeline />
      </InventoryContext.Provider>
    )

    expect(queryByText(/detailLabel/)).toBeNull();
  });
});
