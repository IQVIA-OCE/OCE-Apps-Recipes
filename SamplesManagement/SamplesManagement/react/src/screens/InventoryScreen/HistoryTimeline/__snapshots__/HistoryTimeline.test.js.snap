// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryTimeline should render properly 1`] = `
<View
  style={
    Object {
      "borderRightColor": "#DADADA",
      "borderRightWidth": 1,
      "flexGrow": 1,
      "paddingHorizontal": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "borderBottomColor": "#d8dde6",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 15,
        "paddingBottom": 10,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#4D4D4D",
          }
        }
      >
        label
         ( 
        detailLabel
         )
      </Text>
    </View>
    <View>
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessibilityTraits="button"
        accessible={true}
        centered={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 15,
                  "height": 30,
                  "width": 30,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 20,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "lotId": "1",
          "lotNumber": "",
          "productId": "1",
          "productName": "",
          "quantity": "",
          "recordType": Object {
            "DeveloperName": "AcknowledgementOfShipment",
            "Name": "",
          },
          "transactionDateTime": "",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "padding": 10,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginRight": 12,
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "backgroundColor": "#ef6e64",
                    "borderRadius": 4,
                    "height": 24,
                    "marginBottom": 2,
                    "padding": 3,
                    "width": 24,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 17,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#ef6e64",
                    "flex": 1,
                    "width": 2,
                  }
                }
              />
            </View>
            <View>
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "color": "#080707",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "marginBottom": 5,
                  }
                }
              >
                
              </Text>
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "#080707",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "marginBottom": 5,
                  }
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "color": "#3e3e3c",
                    "fontSize": 11,
                    "fontWeight": "400",
                    "marginBottom": 24,
                  }
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flexBasis": 0,
                "flexGrow": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#3e3e3c",
                  "fontSize": 12,
                }
              }
            >
              
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#f88962",
                  "borderRadius": 50,
                  "padding": 2,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontWeight": "bold",
                  }
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`HistoryTimeline should render properly 2`] = `
<View
  style={
    Object {
      "borderRightColor": "#DADADA",
      "borderRightWidth": 1,
      "flexGrow": 1,
      "paddingHorizontal": 5,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "lotId": "1",
          "lotNumber": "",
          "productId": "1",
          "productName": "",
          "quantity": "",
          "recordType": Object {
            "DeveloperName": "AcknowledgementOfShipment",
            "Name": "",
          },
          "transactionDateTime": "",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "padding": 10,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginRight": 12,
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "backgroundColor": "#ef6e64",
                    "borderRadius": 4,
                    "height": 24,
                    "marginBottom": 2,
                    "padding": 3,
                    "width": 24,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 17,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#ef6e64",
                    "flex": 1,
                    "width": 2,
                  }
                }
              />
            </View>
            <View>
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "color": "#080707",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "marginBottom": 5,
                  }
                }
              >
                
              </Text>
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "#080707",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "marginBottom": 5,
                  }
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "color": "#3e3e3c",
                    "fontSize": 11,
                    "fontWeight": "400",
                    "marginBottom": 24,
                  }
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flexBasis": 0,
                "flexGrow": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#3e3e3c",
                  "fontSize": 12,
                }
              }
            >
              
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#f88962",
                  "borderRadius": 50,
                  "padding": 2,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontWeight": "bold",
                  }
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
